Report only files specified by mypy config

ivirabyan opened the pull request:
Add a setting to  check only a subset of files in the workspace folder. The actual list of files is specified in the [files](https://mypy.readthedocs.io/en/stable/config_file.html#confval-files) mypy configuration.

We omit path argument to mypy, so that it uses files defined in the configuration.

How to manually verify the feature:
1. Create a project with 2 python files: `foo.py` and `bar.py`, each containing typing error.
2. In the project root create a `pyproject.toml` with the following contents:
```
[tool.mypy]
files = ["foo.py"]
```
3. In the project configuration, set `mypy-type-checker.reportingScope` to `custom`.
4. Open `foo.py`, it should have a typing error highlighted
5. Open `bar.py`, it should NOT have typing errors highlighted

Closes https://github.com/microsoft/vscode-mypy/issues/353

karthiknadig commented:
@ivirabyan Thanks for the PR. This is a nice addition. 

I created https://github.com/microsoft/vscode-mypy/issues/353 to track verification of this feature. Can you add steps to manually verify this feature? Another, request is could you add tests for this? you can find `workspace` test case here https://github.com/microsoft/vscode-mypy/blob/7b9ee5534d8c2837c2d4203a02478d7303168c28/src/test/python_tests/test_linting.py#L384 . This seems similar to that.

karthiknadig reviewed (changes requested):
Please add tests here. https://github.com/microsoft/vscode-mypy/blob/7b9ee5534d8c2837c2d4203a02478d7303168c28/src/test/python_tests/test_linting.py

Similar test: https://github.com/microsoft/vscode-mypy/blob/7b9ee5534d8c2837c2d4203a02478d7303168c28/src/test/python_tests/test_linting.py#L384

ivirabyan commented:
Updated description with verification steps.
Also added unittest in https://github.com/microsoft/vscode-mypy/pull/352/commits/4d84ab37a15a2cea50327b9dbfd6b9aadf2d75e0. It was a bit trickier, since I have to prove additional diagnostics _don't_ come, so I've put an extra timeout of 3 seconds there to wait for any additional diagnostics.
